### ISSUE TITLE
Project 7: Improve site architecture and add features to Orcamap

### ISSUE THREAD

**Issue by ivanoats** [orcasound/orcagsoc#9]
7) Add features to the Orcamap

Bring the [orcamap](https://github.com/orcasound/orcamap) features over to a new repo [orcamap-react](https://github.com/orcasound/orcamap-react) in order to improve code organization, testability, and integration with the orcasite repo.

When an orca call is detected through the Orcasound web app (by humans and/or machines), visualize it spatially by plotting it on a dynamic map.

Orcasound community member Christian Sarason developed an open-source repository that contains code to create a Map-box based interface that can both accept user input (a series of points where orcas have been seen or heard) and display locations as confirmed after they have been vetted through by experts via a Google spreadsheet. As tracked in the Orcamap development Trello board, there are three top-requested features could be added this summer: hydrophone locations that “light up” when orcas are being heard via the Orcasound web app; a scale bar to aid accurate input of sighting data; and temporally-filtered views (one of which could manifest as a dynamic map to replace the static ones currently within the v2 UI Orcasound web app). A stretch goal could be to visualize the movement of whales (e.g. by connecting dots with arrows) and the likely predicted locations (e.g. with a circle that grows according to the mean swim speed or orcas).

Required skills: Node, Javascript, Google sheets API, ReactJS
Bonus skills: MapBox, Open Layers
Possible mentors:  [Ivan](https://github.com/ivanoats), Scott and Val (with technical assistance from Orcasound community member Christian)
References and open-source building blocks:
Orcasound Github issue for this project
[Orcasound/orcamap-react](https://github.com/orcasound/orcamap-react) repository

**Comment by azanbinzahid** [comment]
Please update the issue link (to this) in Project Ideas doc.

**Comment by ivanoats** [comment]
@azanbinzahid done

**Comment by riteshsp2000** [comment]
Hey @ivanoats @azanbinzahid. I would like to work on this. Any plan of action as of now?

**Comment by Devesh21700Kumar** [comment]
i would like to take this one.. really excited to contribute here in GSOC 2021

**Comment by Devesh21700Kumar** [comment]
will we be given permissions to the google sheets key? i guess we need to add it in our env variables

**Comment by ivanoats** [comment]
We need to think of a better security policy if there are several contributors. For now, could you use your own google ID and google sheets API key? I think we would need an example sheet.

**Comment by Devesh21700Kumar** [comment]
> We need to think of a better security policy if there are several contributors. For now, could you use your own google ID and google sheets API key? I think we would need an example sheet.

yeah figured so. I've created one for myself and have made a PR. Please do check it out

**Comment by arnab15** [comment]
@ivanoats I'm Arnab Sahoo, Currently pursuing undergraduate studies  I'm a MERN stack developer and open-source  Enthusiast. I'm very excited to contribute here to GSOC 2021.

**Comment by ivanoats** [comment]
@arnab15 Please see https://github.com/orcasound/orcamap-react/blob/main/CONTRIBUTING.md and https://github.com/orcasound/orcamap-react/issues/23

**Comment by imacchiavello** [comment]
i am  really excited to be able to contribute  here in GSOC 2021

**Comment by scottveirs** [comment]
@ivanoats @cpsarason I've added a few links in project description to help provide context, history, and the beginnings of a longer-range road map for the GSoC students. Also added a link to the prototype deployed via netlify.
